### DISCUSSION TITLE
When root loader gets called?

### DISCUSSION THREAD

**Discussion by shanmugharajk** [remix-run/react-router#11611] [Q&A] [answered]
Hi team,

I'm trying to find out under what conditions root loaders gets called? I somehow couldn't find anything related to this in doc.

Right now I have a setup like this
   - action employee.create.ts
   - route employee_.index.tsx
        - `<Form action="/employee/create" />`
 
When I submit the form, after the action gets invoked the `loader` in root also gets called. Can I rely on this behaviour that root `loader` gets every time on this form submit?

**Comment by sergiodxa** [comment, marked as answer]
When the browser sends a document request and the server respond with html, the root loader is called to do the SSR of the app

After an action run, Remix trigger revalidation of all loaders, this can be prevented on a per route basis using the shouldRevalidate export.

When the search params change (the part of the URL after `?`), because any route can use them, then all loaders run again, including root.

If you enable the Single Fetch future flag, any client side navigation will run fetch every loader.
